yksgi/src/lib.rs

```diff
@@ -395,6 +431,14 @@ impl SGInterp {
         }
     }
 
+    fn ptrtoint(&mut self) {
+        let src = self.pc.get_operand(0);
+        let srcty = src.get_type();
+        assert!(srcty.is_pointer());
+        let srcval = self.var_lookup(&src).val;
+        self.var_set(self.pc, SGValue::new(srcval, self.pc.get_type()));
```

Comment:
You can probably use `SGValue::with_type` here, which creates a copy with a new type.
```
let srcval = self.var_lookup(&src);
self.var_set(self.pc, SGValue::with_type(self.pc.get_type()));
```

That being said, this will all go away when we get rid of the type within `SGValue`.